pkg/cmd/join.go

```diff
@@ -136,7 +137,13 @@ func setupAgent(serverIP, ip net.IP, port int, user, sshKeyPath, joinToken strin
 
 	defer operator.Close()
 
-	getTokenCommand := fmt.Sprintf(`curl -sfL https://get.k3s.io/ | K3S_URL="https://%s:6443" K3S_TOKEN="%s" sh -`, serverIP.String(), strings.TrimSpace(joinToken))
+	if !useDocker {
+		getTokenCommand := fmt.Sprintf(`curl -sfL https://get.k3s.io/ | K3S_URL="https://%s:6443" K3S_TOKEN="%s" sh -`, serverIP.String(), strings.TrimSpace(joinToken))
+	}
+	else {
+		getTokenCommand := fmt.Sprintf(`curl -sfL https://get.k3s.io/ | INSTALL_K3S_EXEC="--docker" K3S_URL="https://%s:6443" K3S_TOKEN="%s" sh -`, serverIP.String(), strings.TrimSpace(joinToken))
+	}
```

Comment:
I'd rather see the string constructed with fmt.Sprintf and have a single if statement to populate the options (if needed)

Reply:
Can you acknowledge the comment when you have time please?